src/linear_algebra/matrix/circulant.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import linear_algebra.matrix.symmetric
+import data.polynomial.monomial
+import data.matrix.pequiv
+import data.equiv.fin
+
+/-!
+# Circulant matrices
+
+This file contains the definition and basic results about circulant matrices.
+
+## Main results
+
+- `matrix.cir`: introduce the definition of a circulant matrix
+                generated by a given vector `v : I → α`.
+
+## Implementation notes
+
+`fin.foo` is the `fin n` version of `foo`.
+Namely, the index type of the circulant matrices in discussion is `fin n`.
+
+## Tags
+
+cir, matrix
+-/
+
+variables {α β I J R : Type*} {n : ℕ}
+
+namespace matrix
+open_locale matrix big_operators
+
+/-- Given the condition `[has_sub I]` and a vector `v : I → α`,
+    we define `cir v` to be the circulant matrix generated by `v` of type `matrix I I α`. -/
+def cir [has_sub I] (v : I → α) : matrix I I α
+| i j := v (i - j)
+
+lemma cir_col_zero_eq [add_group I] (v : I → α) :
+  (λ i, (cir v) i 0) = v :=
+by ext; simp [cir]
+
+lemma cir_ext_iff [add_group I] {v w : I → α} :
```

Comment:
```suggestion
lemma cir_inj [add_group I] {v w : I → α} :
```
But you should also prove `cir_injective : function.injective (cir : (I → α) → _)

Reply:
This comment still applies

Reply:
I am not sure if I have correctly understood the suggestion here. Do you mean to change the identifier or add a new lemma?

Reply:
If it's to change the identifier, isn't `cir_ext_iff` consistent with other identifiers used for matrices related lemmas in mathlib?

Reply:
@eric-wieser

Reply:
I'm asking you to rename this lemma to `*_inj`, because that's what the naming guide suggests.